core/app/models/spree/product.rb

```diff
@@ -105,7 +105,7 @@ class Product < Spree::Base
       validates :price, if: proc { Spree::Config[:require_master_price] }
     end
 
-    validates :slug, length: { minimum: 3 }, allow_blank: true, uniqueness: true
+    validates :slug, allow_blank: true, uniqueness: true
```

Comment:
@krzysiek1507 turns out the `allow_blank` is there to remove the extra query being made to check if the slug is unique when it's blank.
https://github.com/spree/spree/pull/5565

I'd keep it, but maybe add `presence: true` in the mix as well. What do you think? @damianlegawiec @krzysiek1507